api/verify.go

```diff
@@ -1,177 +0,0 @@
-func (c *Context) VerifyKZGProof(blobCommitment serialization.KZGCommitment, kzgProof serialization.KZGProof, inputPointBytes, claimedValueBytes serialization.Scalar) error {
-	// 1. Deserialization
-	//
-	claimedValue, err := serialization.DeserializeScalar(claimedValueBytes)
-	if err != nil {
-		return err
-	}
-
-	inputPoint, err := serialization.DeserializeScalar(inputPointBytes)
-	if err != nil {
-		return err
-	}
-
-	polynomialCommitment, err := serialization.DeserializeG1Point(blobCommitment)
-	if err != nil {
-		return err
-	}
-
-	quotientCommitment, err := serialization.DeserializeG1Point(kzgProof)
-	if err != nil {
-		return err
-	}
-
-	// 2. Verify opening proof
-	proof := kzg.OpeningProof{
-		QuotientCommitment: quotientCommitment,
-		InputPoint:         inputPoint,
-		ClaimedValue:       claimedValue,
-	}
-
-	return kzg.Verify(&polynomialCommitment, &proof, c.openKey)
-}
-
-// [verify_blob_kzg_proof](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#verify_blob_kzg_proof)
-func (c *Context) VerifyBlobKZGProof(blob serialization.Blob, blobCommitment serialization.Commitment, kzgProof serialization.KZGProof) error {
-	// 1. Deserialize
-	//
-	polynomial, err := serialization.DeserializeBlob(blob)
-	if err != nil {
-		return err
-	}
-
-	polynomialCommitment, err := serialization.DeserializeG1Point(blobCommitment)
-	if err != nil {
-		return err
-	}
-
-	quotientCommitment, err := serialization.DeserializeG1Point(kzgProof)
-	if err != nil {
-		return err
-	}
-
-	// 2. Compute the evaluation challenge
-	evaluationChallenge := computeChallenge(blob, blobCommitment)
-
-	// 3. Compute output point/ claimed value
-	outputPoint, err := c.domain.EvaluateLagrangePolynomial(polynomial, evaluationChallenge)
-	if err != nil {
-		return err
-	}
-
-	// 4. Verify opening proof
-	openingProof := kzg.OpeningProof{
-		QuotientCommitment: quotientCommitment,
-		InputPoint:         evaluationChallenge,
-		ClaimedValue:       *outputPoint,
-	}
-
-	return kzg.Verify(&polynomialCommitment, &openingProof, c.openKey)
-}
-
-// [verify_blob_kzg_proof_batch](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#verify_blob_kzg_proof_batch)
-func (c *Context) VerifyBlobKZGProofBatch(blobs []serialization.Blob, polynomialCommitments serialization.Commitments, kzgProofs []serialization.KZGProof) error {
-	// 1. Check that all components in the batch have the same size
-	//
-	blobsLen := len(blobs)
-	lengthsAreEqual := blobsLen == len(polynomialCommitments) && blobsLen == len(kzgProofs)
-	if !lengthsAreEqual {
-		return ErrBatchLengthCheck
-	}
-	batchSize := blobsLen
-
-	// 2. Collect opening proofs
-	//
-	openingProofs := make([]kzg.OpeningProof, blobsLen)
```

Comment:
I think it would be cleaner if `blobsLen` is `batchSize` here and in the line below.